### DISCUSSION TITLE
Help with notifications on events like player disconnects

### DISCUSSION THREAD

**Discussion by SmutsMuts** [lloesche/valheim-server-docker#397] [Q&A]
There is an example for how to make a notification when a player spawns in a server, are there any other fun examples of notifications I can get some help in making? 
I would like to have a notification when someone disconnects from the server.
I would like to have a notification when someone dies. 
I would like to have a notification when an event like trolls attacking or similar is happening in the game.
Could someone help me with this?  

What I was refering to earlier was: 
`-e DISCORD_WEBHOOK="https://discord.com/api/webhooks/8171522530..." \`
`-e VALHEIM_LOG_FILTER_CONTAINS_Spawned="Got character ZDOID from" \`
`-e ON_VALHEIM_LOG_FILTER_CONTAINS_Spawned='{ read l; l=${l//*ZDOID from /}; l=${l// :*/}; msg="Player $l spawned into the world"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }'`

**Comment by smosse** [comment]
@SmutsMuts  Did you make this one work ? im not sure about the "Got character ZDOID from"

**Reply by AlfaJackal** [reply]
Any hints here what to do? I am also interested in sending a message to Discord when a player disconnects.

**Reply by PyroghostX** [reply]
I had to use regex to get it to work

VALHEIM_LOG_FILTER_REGEXP_Spawned=\bGot character ZDOID from \b\w*\b : [^0]\d*\b:2\b
ON_VALHEIM_LOG_FILTER_REGEXP_Spawned={ read l; l=${l//*ZDOID from /}; l=${l// :*/}; msg="Player $l is online!"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }

VALHEIM_LOG_FILTER_REGEXP_Death=\bGot character ZDOID from \b\w*\b : 0:0
ON_VALHEIM_LOG_FILTER_REGEXP_Death={ read l; l=${l//*ZDOID from /}; l=${l// :*/}; msg="$l just died!"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }

VALHEIM_LOG_FILTER_CONTAINS_RandomEvent=Random event
ON_VALHEIM_LOG_FILTER_CONTAINS_RandomEvent={ read l; l=${l//*event set: /}; l=${l// :*/}; msg="Random event: $l"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }

**Reply by jonvel** [reply]
I was only able to map a Logoff event from pushing the `ZDOID` (the `\d+:\d+` in the `Got character ZDOID from`, as long as that `ZDOID` isn't `0:0` - which TIL is "character died" ;) ) onto a stack, and then when you get a `Destroying abandoned non persistent zdo` event with that matching `ZDOID` (whatever you want to call it).  Which means you can't actually parse it in a vacuum, as you need to have a running list/hash of currently logged in users mapped to their current `ZDOID`.  At least, that's what I'd do.  But that's speculation and it kind of works.  I don't push that event to a Discord server, though.  That's too hard.
